drivers/adc/adc_npcx.c

```diff
@@ -93,8 +90,9 @@ static void adc_npcx_isr(void *arg)
 		/* Get result for each ADC selected channel */
 		while (data->channels) {
 			channel = find_lsb_set(data->channels) - 1;
-			result = GET_FIELD(CHNDAT(DRV_CONFIG((const struct device *)arg)->base,
-						  channel),
+			result = GET_FIELD(
+				CHNDAT(((const struct adc_npcx_config *)dev->config)->base,
```

Comment:
nit: for better readability, dereference config first

```c
const struct adc_npcx_config *config = dev->config;
...
result = GET_FIELD(CHNDAT(config->base, channel), NPCX_CHNDAT_CHDAT_FIELD);
```

Reply:
done.